File: 15-18/HotFix/HotFixDemo/app/src/main/java/com/wx/hotfixdemo/MainActivity.java
package com.wx.hotfixdemo;

import android.Manifest;
import android.databinding.ViewDataBinding;
import android.view.View;
import android.widget.Toast;

import com.tbruyelle.rxpermissions.RxPermissions;
import com.wx.hotfixdemo.base.BaseActivity;
import com.wx.hotfixdemo.databinding.ActivityMainBinding;

import rx.functions.Action1;

public class MainActivity extends BaseActivity {

    private ActivityMainBinding binding;

    @Override
    public int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    public void setDataBinding(ViewDataBinding binding) {
        this.binding = (ActivityMainBinding) binding;
    }

    @Override
    public void initData() {
        initPermission();
        binding.btnClick.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //SophixManager.getInstance().queryAndLoadNewPatch();
                //Toast.makeText(MainActivity.this, "测试版本:"+ VersionTool.getVersionName(MainActivity.this),Toast.LENGTH_SHORT).show();
                //Toast.makeText(MainActivity.this, "升级版本:"+VersionTool.getVersionName(MainActivity.this),Toast.LENGTH_SHORT).show();
                Toast.makeText(MainActivity.this, "第三个HotFix",Toast.LENGTH_SHORT).show();
            }
        });
    }

    public void initPermission() {
        String[] permissions = new String[]{Manifest.permission.READ_EXTERNAL_STORAGE,Manifest.permission.ACCESS_WIFI_STATE};
        RxPermissions rxPermissions = new RxPermissions(this);
        rxPermissions.request(permissions)
                .subscribe(new Action1<Boolean>() {
                    @Override
                    public void call(Boolean aBoolean) {
                        if (!aBoolean) {
                            Toast.makeText(MainActivity.this,"请添加权限，否则后续测试可能失败",Toast.LENGTH_SHORT).show();
                        }
                    }
                });
    }

}

File: Android-Demo/BaseMaster1/app/src/main/java/com/wx/basemaster1/MainActivity.kt
package com.wx.basemaster1

import android.databinding.ViewDataBinding
import com.wx.basemaster1.activity.ContrainLayout1Activity
import com.wx.basemaster1.activity.ContrainLayoutActivity
import com.wx.basemaster1.activity.KotlinActivity
import com.wx.basemaster1.base.BaseActivity
import com.wx.basemaster1.databinding.ActivityMainBinding

class MainActivity : BaseActivity() {

    private var binding: ActivityMainBinding? = null

    override val layoutId: Int
        get() = R.layout.activity_main

    override fun setDataBinding(binding: ViewDataBinding) {
        this.binding = binding as ActivityMainBinding
    }

    override fun initData() {
        binding!!.btnConstrain.setOnClickListener { openActivity(ContrainLayoutActivity::class.java) }
        binding!!.btnConstrain1.setOnClickListener { openActivity(ContrainLayout1Activity::class.java) }
        binding!!.btnKotlin.setOnClickListener { openActivity(KotlinActivity::class.java) }
        binding!!.btnNetwork.setOnClickListener { openActivity(KotlinActivity::class.java) }
    }

}
